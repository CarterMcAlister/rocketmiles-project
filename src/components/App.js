import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

import Nav from './Nav';
import FilterSidebar from './FilterSidebar';
import HotelList from './HotelList';
import Error from './Error';

import { breakpoints } from '../styleVariables';
import hotelResultService from '../services/hotel-result/hotel-result.service';

const App = () => {
    const [hotels, setHotels] = useState([]);
    const [sortOrder, setSortOrder] = useState('recommended');
    const [nameFilter, setNameFilter] = useState('');
    const [showResults, setShowResults] = useState(true);
    const [isLoading, setIsLoading] = useState(true);
    const [errorMsg, setErrorMsg] = useState('Sorry, something went wrong. Please try again.');

    // Get hotel info and display, or show error message
    const getHotelInfo = () => {
        setIsLoading(true);
        hotelResultService
            .get()
            .then(response => {
                if (response && response.success) {
                    setHotels(response.results.hotels);
                    setShowResults(response.success);
                } else {
                    setShowResults(false);
                    setErrorMsg(response.message);
                }
            })
            .finally(() => {
                setIsLoading(false);
            });
    };

    useEffect(() => {
        getHotelInfo();
    }, []);

    return (
        <div className="app-container">
            <Nav />
            {showResults ? (
                <ContentLayout>
                    <FilterSidebar
                        sortOrder={sortOrder}
                        setSortOrder={setSortOrder}
                        nameFilter={nameFilter}
                        setNameFilter={setNameFilter}
                    />
                    <HotelList hotels={hotels} sortOrder={sortOrder} nameFilter={nameFilter} isLoading={isLoading} />
                </ContentLayout>
            ) : (
                <Error message={errorMsg} handleClick={getHotelInfo} />
            )}
        </div>
    );
};

const ContentLayout = styled.div`
    margin: 0 auto;
    background: #dcdcdc;

    @media ${breakpoints.small} {
        display: grid;
        grid-template-columns: 20% 75%;
        grid-column-gap: 20px;
        max-width: 1200px;
        padding: 10px;
    }
`;

export default App;
