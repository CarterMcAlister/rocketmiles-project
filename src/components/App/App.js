import React, { useState, useEffect } from 'react';
import './App.style.scss';

import Filter from '../Filters';
import HotelList from '../HotelList';

import hotelResultService from '../../services/hotel-result/hotel-result.service';

const App = () => {
    const [hotels, setHotels] = useState([]);

    useEffect(() => {
        hotelResultService.get().then(response => {
            setHotels(response.results.hotels);
        });
    }, []);

    return (
        <div className="app-container">
            <div className="content">
                <Filter />
                <HotelList hotels={hotels} />
            </div>
        </div>
    );
};

export default App;
