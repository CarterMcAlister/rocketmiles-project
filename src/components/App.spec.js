import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
    it('renders the component', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('.app-container').exists()).toBe(true);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
