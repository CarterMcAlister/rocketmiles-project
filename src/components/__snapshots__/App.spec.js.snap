// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the component 1`] = `"<div class=\\"app-container\\"><div class=\\"sc-bdVaJa hIGtMT\\"><img src=\\"https://www.rocketmiles.com/resources/images/logos/rocketmiles-two-tone-logo.png\\" class=\\"sc-bwzfXH iAtoAT\\"/></div><div class=\\"sc-cSHVUG gtKVBd\\"><div class=\\"sc-htpNat iMfrpM\\"><div class=\\"filters\\">Hotel name<input type=\\"text\\" class=\\"input\\" value=\\"\\"/>Price<select name=\\"\\" class=\\"select\\"><option selected=\\"\\" value=\\"recommended\\">Recommended</option><option value=\\"low-to-high\\">Price low-to-high</option><option value=\\"high-to-low\\">Price high-to-low</option></select><button class=\\"button\\">Reset</button></div></div><div class=\\"sc-gqjmRU cQRTIW\\"><div class=\\"sc-gZMcBi cAFfKU\\"><div class=\\"la-line-scale la-2x\\"><div></div><div></div><div></div><div></div><div></div></div></div></div></div></div>"`;
